### PR TITLE
Added collapsable breadcrumb

### LINKED ISSUE #427: Truncate long breadcrumbs
Long breadcrumbs aren't getting truncated and are cut off by the Parts/Guides/Answers element
https://es.ifixit.com/Parts/MacBook_Pro_17%22_Unibody

image

### PR DESCRIPTION
**Pull request by rickisXP**
Closes #427 

Summary
--
Previously, breadcrumbs were not being collapsed on product list pages, which led to really long breadcrumbs. This pull aims to fix that by porting over some of the `PageBreadcrumb.tsx` code.

QA
--
Check that on product pages, if there isn't enough room to fit the entire breadcrumb trail, it collapses sensibly.

### CODE REVIEW

**Comment by deltuh-vee** [comment]
dev_block ✊ 
I think the previous breadcrumb link is missing from the collapsed menu breadcrumbs versus the fully expanded breadcrumbs.
image

image

**Comment by rickisXP** [comment]
un_dev_block :star2:

**Comment by deltuh-vee** [comment]
QA 🎬 
Product pages with more than 3 breadcrumbs collapse into a menu.

**Comment by rickisXP** [comment]
un_dev_block :star: and deploy_block :thought_balloon: on the following question:

What should be done about the ProductListBreadcrumb.tsx file? It seems like there aren't anymore references to this component.

File: frontend/templates/product-list/SecondaryNavigation.tsx [hunk 1]
```diff
@@ -42,12 +53,14 @@ export function SecondaryNavigation({ productList }: SecondaryNavigationProps) {
...
                >
-                  <ProductListBreadcrumb
+                  <BreadCrumbs
                      display={{
                         base: 'none',
                         sm: 'flex',
                      }}
-                     productList={productList}
+                     breadCrumbs={breadCrumbs}
+                     breadcrumbsToShow={3}
```
Comment (jarstelfox) [inline frontend/templates/product-list/SecondaryNavigation.tsx:62]: Minor: I think we may be comfortable allowing the breadcrumbs to be dynamic (size to the available space)

**Comment by deltuh-vee** [comment]
dev_block ✊ on merge conflicts

File: frontend/templates/product-list/SecondaryNavigation.tsx (hunk 1 above)
Comment (sterlinghirsh) [reply frontend/templates/product-list/SecondaryNavigation.tsx:62]: Fine with me

**Comment by erinemay** [comment]
It looks like @ianrohde is working on breadcrumb collapsibility over here https://github.com/iFixit/react-commerce/pull/1635. Can anything be shared?

**Comment by rickisXP** [comment]
un_dev_block :star: but deploy_block :thought_balloon: on the [question above ](https://github.com/iFixit/react-commerce/pull/1586#issuecomment-1530572820)

File: frontend/templates/product-list/SecondaryNavigation.tsx [hunk 2]
```diff
@@ -19,6 +21,15 @@ export function SecondaryNavigation({ productList }: SecondaryNavigationProps) {
...
    const hasDeviceNavigation =
       productList.type !== ProductListType.AllParts && !isToolsProductList;
+   const { currentItemTitle, ancestors } = useProductListAncestors(productList);
+   const breadCrumbs: BreadcrumbItem[] = ancestors.map((ancestor) => ({
+      label: ancestor.title,
+      url: productListPath(ancestor),
+   }));
+   breadCrumbs.push({
+      label: currentItemTitle,
+      url: undefined,
```
Comment (jarstelfox) [inline frontend/templates/product-list/SecondaryNavigation.tsx:31]: This adds clarity but is unneeded. Up to you if you leave it.

```suggestion
```

Comment (jarstelfox) [reply frontend/templates/product-list/SecondaryNavigation.tsx:31]: We also could make this link the to current page or generate the correct link for the current page?

**Review by jarstelfox — COMMENTED** [review]
CR 🌵

**Review by jarstelfox — APPROVED** [review]

**Comment by k0rvusk0r4x** [comment]
Working nicely as above.

QA 😺

**Comment by jarstelfox** [comment]
> What should be done about the ProductListBreadcrumb.tsx file? It seems like there aren't anymore references to this component, so would it be safe to delete?

I'd vote to remove it, but I'd love someone from the Shopify squad to give the go ahead

**Comment by dhmacs** [comment]
@jarstelfox Drop it 👍

**Comment by rickisXP** [comment]
un_deploy_block :star:

**Review by jarstelfox — APPROVED** [review]
CR 🌵

**Pull request merged**